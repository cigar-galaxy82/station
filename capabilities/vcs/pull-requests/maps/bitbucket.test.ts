import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`vcs/pull-requests/bitbucket`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('PullRequests', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'vcs/pull-requests',
          provider: 'bitbucket',
          useCase: 'PullRequests',
          input: {
            owner: 'JakubVacek',
            repo: 'test',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
  it('should handle error', async () => {
    await expect(
      superface.run({
        profile: 'vcs/pull-requests',
        provider: 'bitbucket',
        useCase: 'PullRequests',
        input: {
          owner: 'JakubVacek',
          repo: 'made-up',
        },
      })
    ).resolves.toMatchSnapshot();
  });
});
