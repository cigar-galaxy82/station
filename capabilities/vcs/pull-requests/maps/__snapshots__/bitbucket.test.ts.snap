// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-requests/bitbucket PullRequests should perform successfully 1`] = `
Ok {
  "value": Object {
    "pullRequests": Array [
      Object {
        "id": 22,
        "sha": "d28a583d772c",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/22",
      },
      Object {
        "id": 21,
        "sha": "1eca58efb8a9",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/21",
      },
      Object {
        "id": 20,
        "sha": "afe1bc038f22",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/20",
      },
      Object {
        "id": 19,
        "sha": "1ac2fe089fd4",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/19",
      },
      Object {
        "id": 18,
        "sha": "097ee8a09c8c",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/18",
      },
      Object {
        "id": 17,
        "sha": "e761e1ae0d95",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/17",
      },
      Object {
        "id": 16,
        "sha": "01d8b9a26d75",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/16",
      },
      Object {
        "id": 15,
        "sha": "44e23928eda0",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/15",
      },
      Object {
        "id": 14,
        "sha": "e47f0b79a36e",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/14",
      },
      Object {
        "id": 13,
        "sha": "e5bdf026db1f",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/13",
      },
      Object {
        "id": 12,
        "sha": "7a3bdb34a4a1",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/12",
      },
      Object {
        "id": 11,
        "sha": "87cd8d7b07bc",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/11",
      },
      Object {
        "id": 10,
        "sha": "11327596c343",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/10",
      },
      Object {
        "id": 9,
        "sha": "ac9d6d6e1fe8",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/9",
      },
      Object {
        "id": 8,
        "sha": "4a86529bde0b",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/8",
      },
      Object {
        "id": 7,
        "sha": "56ac7fabf85e",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/7",
      },
      Object {
        "id": 6,
        "sha": "77e1bab364a6",
        "title": "README.md edited online with Bitbucket 6",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/6",
      },
      Object {
        "id": 5,
        "sha": "b15e30046c66",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/5",
      },
      Object {
        "id": 4,
        "sha": "f32b5c7e01db",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/4",
      },
      Object {
        "id": 3,
        "sha": "105821b55f10",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/3",
      },
      Object {
        "id": 2,
        "sha": "d24e5bb8ba8e",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/2",
      },
      Object {
        "id": 1,
        "sha": "631e67580509",
        "title": "README.md edited online with Bitbucket",
        "url": "https://bitbucket.org/JakubVacek/test/pull-requests/1",
      },
    ],
    "statusCode": 200,
  },
}
`;

exports[`vcs/pull-requests/bitbucket should handle error 1`] = `
Err {
  "error": "MappedError: Expected error
Properites: {
  \\"message\\": \\"The requested repository either does not exist or you do not have access. If you believe this repository exists and you have access, make sure you're authenticated.\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[2].statements[0]
Original Map Location: Line 15, column 7",
}
`;
